src/script/script.cpp

```diff
@@ -198,20 +198,57 @@ unsigned int CScript::GetSigOpCount(const CScript& scriptSig) const
     return subscript.GetSigOpCount(true);
 }
 
+bool CScript::IsPayToPublicKey() const
+{
+    // Test for pay-to-public-key CScripts:
+    return (((this->size() == 0x43 && (*this)[0] == 0x41) ||
+            (this->size() == 0x23 && (*this)[0] == 0x21)) &&
+            (*this)[1] == OP_EQUALVERIFY);
```

Comment:
In 397f7928736e55acd81857ad95cbcd47f0fd569b:

This doesn't look correct. p2pk outputs have no `OP_EQUALVERIFY` but they do have a `OP_CHECKSIG`. And this needs to occur after the public key.

Reply:
Fixed this and modified the code to accept the correct PUBKEY scripts, The TxoutType::PUBKEY converts the output script to PUBKEYHASH by default. Thanks!